### DISCUSSION TITLE
Why is my binding not updating?

### DISCUSSION THREAD

**Discussion by Logopher** [AvaloniaUI/Avalonia#17221] [Q&A]
Using ReactiveUI, I made a read-write property for a binding:
    
        private string _test = "N/A";

        public string Test
        {
            get => _test;
            set => this.RaiseAndSetIfChanged(ref _test, value);
        }

It's updated when arrow keys are pressed:

        private void Window_KeyDown(object? sender, KeyEventArgs e)
        {
            switch (e.Key)
            {
                case Key.Up:
                    VM.Test = "Up";
                    break;
                case Key.Down:
                    VM.Test = "Down";
                    break;
                case Key.Left:
                    VM.Test = "Left";
                    break;
                case Key.Right:
                    VM.Test = "Right";
                    break;
            }
        }

Using Subscribe or breakpoints, I can see that the property is being updated, but the UI never changes.

I did something similar in a different project, and had no problems. I haven't been able to identify a difference, so I reduced the problem project, and attached it: [BindingTest.zip](https://github.com/user-attachments/files/17297467/BindingTest.zip)

Any idea what's going on?

**Comment by IanRawley** [comment]
One problem I can see, although I'd think it would cause compilation errors with Compiled Bindings, is that MainWindow.axaml has the vm namespace as BunBun.ViewModels while the VM itself is in BindingTest.ViewModels

**Comment by IanRawley** [comment]
Found the real problem.  Your keyhandler is changing the Test property on the MainWindow's VM property, but the actual view is bound to the MainWindow's DataContext.  You have two MainViewModel objects.  Get rid of the private VM property on MainWindow, or change it to refer to the DataContext with a cast.
